AnalysisNtuple/Selector.cpp

```diff
@@ -162,6 +177,41 @@ void Selector::filter_photons(){
         double absEta = TMath::Abs(eta);
         double phi = tree->phoPhi_[phoInd];
 
+
+
+
+
+        double PhoSmear = 1.;
+
+        if (!tree->isData_ && (phosmearLevel==0 || elesmearLevel==2)) {
+        	if (et <= 15) {et = 16.0;}
+        	if (absEta >= 2.4) {absEta = 2.39;}
+        	PhoSmear = getElePhoSmearScaleSF(et,absEta); 
```

Comment:
You should not be updating the actual et or absEta variables, but temporary ones that only get used for the lookup.  Also, you need to handle the case where et is greater than the max (this may be even more important than handling the min)

```
double tempEt = et;
double tempAbsEta = absEta;
if (et<15) {tempEt = 15.01;}
if (et>MAXETINHIST) {tempEt = MAXETINHIST-0.1;}
if (absEta >=2.4) {tempAbsEta=2.39;}
...
```